include/emp/Evolve/Systematics.hpp

```diff
@@ -921,490 +1124,205 @@ namespace emp {
-    Ptr<taxon_t> AddOrg(ORG && org, Ptr<taxon_t> parent=nullptr, int update=-1);
-
-    void AddOrg(ORG & org, WorldPosition pos, int update=-1);
-    Ptr<taxon_t> AddOrg(ORG & org, WorldPosition pos, Ptr<taxon_t> parent=nullptr, int update=-1);
-    Ptr<taxon_t> AddOrg(ORG & org, Ptr<taxon_t> parent=nullptr, int update=-1);
 
-
-    /// Remove an instance of an organism; track when it's gone.
-    bool RemoveOrg(WorldPosition pos, int time=-1);
-    bool RemoveOrg(Ptr<taxon_t> taxon, int time=-1);
-
-    void RemoveOrgAfterRepro(WorldPosition pos, int time=-1);
-    void RemoveOrgAfterRepro(Ptr<taxon_t> taxon, int time=-1);
-
-    /// Remove org from next population (for use with synchronous generations)
-    // bool RemoveNextOrg(WorldPosition pos, int time=-1);
-    // bool RemoveNextOrg(Ptr<taxon_t> taxon, int time=-1);
-
-    /// Climb up a lineage...
-    Ptr<taxon_t> Parent(Ptr<taxon_t> taxon) const;
+    // ===== Output functions ====
 
     /// Print details about the Systematics manager.
+    /// First prints setting, followed by all active, ancestor, and outside
+    /// taxa being stored. Format for taxa is
+    /// [ id | number of orgs in this taxon, number of offspring taxa of this taxon | parent taxon]
+    /// @param os output stream to print to
     void PrintStatus(std::ostream & os=std::cout) const;
 
-    /// Print whole lineage.
+    /// Print a whole lineage. Format: "Lineage:", followed by each taxon in the lineage, each on new line
+    /// @param taxon a pointer to the taxon to print the lineage of
+    /// @param os output stream to print to
     void PrintLineage(Ptr<taxon_t> taxon, std::ostream & os=std::cout) const;
 
+    /// Add a new snapshot function.
+    /// When a snapshot of the systematics is taken, in addition to the default
+    /// set of functions, all user-added snapshot functions are run. Functions
+    /// take a reference to a taxon as input and return the string to be dumped
+    /// in the file at the given key.
+    void AddSnapshotFun(const std::function<std::string(const taxon_t &)> & fun,
+                        const std::string & key, const std::string & desc="") {
+      user_snapshot_funs.emplace_back(fun, key, desc);
+    }
+
+    /// Take a snapshot of current state of taxon phylogeny.
+    /// WARNING: Current, this function assumes one parent taxon per-taxon.
+    /// @param file_path the file to store the snapshot data in
     void Snapshot(const std::string & file_path) const;
 
-    /// Calculate the genetic diversity of the population.
-    double CalcDiversity() const;
+    void SwapPositions(WorldPosition p1, WorldPosition p2) {
+      emp::vector<Ptr<taxon_t> > & v1 = taxon_locations[p1.GetPopID()];
+      emp::vector<Ptr<taxon_t> > & v2 = taxon_locations[p2.GetPopID()];
+      std::swap(v1[p1.GetIndex()], v2[p2.GetIndex()]);
+    }
+
+    void LoadFromFile(const std::string & file_path, const std::string & info_col = "info",
```

Comment:
Added a docstring. The info_col contains the information that taxa are defined based on. So "id" will usually work, but it could more accurately be something like "genotype"